rs/canonical_state/tests/compatibility.rs

```diff
@@ -255,18 +217,11 @@ fn stream_header_encoding_panic_on_invalid(
 /// Produces a `RequestOrResponse` valid at all certification versions in the range.
 pub(crate) fn arb_valid_versioned_message(
 ) -> impl Strategy<Value = (RequestOrResponse, RangeInclusive<CertificationVersion>)> {
-    prop_oneof![
-        (
-            // No `deadline` before version 18.
-            arbitrary::request_or_response_with_config(false),
-            Just(MIN_SUPPORTED_CERTIFICATION_VERSION..=MAX_SUPPORTED_CERTIFICATION_VERSION)
-        ),
-        (
-            // Optionally populate `deadline` from version 18 on.
-            arbitrary::request_or_response_with_config(true),
-            Just(CertificationVersion::V18..=MAX_SUPPORTED_CERTIFICATION_VERSION)
-        ),
-    ]
+    prop_oneof![(
+        // Optionally populate `deadline` from version 18 on.
+        arbitrary::request_or_response_with_config(true),
```

Comment:
I was going to say that we should also get rid of `request_or_response_with_config()` / `request_with_config()` / `response_with_config()`. But since we will be adding the optional `refund_id` field to both `Request` and `Response` soon, I'd rather keep it and repurpose it then.